score/conda.py

```diff
@@ -0,0 +1,137 @@
+        "name", None
+    )
+    required_json_data["versions"] = package_data_response.get("versions", [])
+    required_json_data["latest_version"] = package_data_response.get(
+        "latest_version", None
+    )
+    required_json_data["revision"] = package_data_response.get("revision", None)
+    required_json_data["license"] = package_data_response.get("license", None)
+    required_json_data["license_url"] = package_data_response.get("license_url", None)
+    required_json_data["dev_url"] = package_data_response.get("dev_url", None)
+    required_json_data["doc_url"] = package_data_response.get("doc_url", None)
+    required_json_data["source_git_url"] = package_data_response.get(
+        "source_git_url", None
+    )
+    required_json_data["source_git_tag"] = package_data_response.get(
+        "source_git_tag", None
+    )
+    required_json_data["watchers"] = package_data_response.get("watchers", None)
+    required_json_data["upvoted"] = package_data_response.get("upvoted", None)
+    required_json_data["downloads"] = sum(
+        [i.get("ndownloads", 0) for i in package_data_response.get("files", [])]
+    )
+    required_json_data["package_url"] = package_url
+    return required_json_data
+
+
+def get_package_data(package_name: str) -> dict:
+    """
+    Fetches the package data for the given package name.
+
+    Args:
+        package_name (str): The package name.
+
+    Returns:
+        required_json_data (dict): The required JSON data.
+    """
+    logger.info("Package Name : %s", package_name)
+    package_url = f"https://api.anaconda.org/package/conda-forge/{package_name}"
+    response = requests.get(package_url)
+    if response.status_code != 200:
+        logger.error("Failed to fetch data for package : %s", package_name)
+        return None
+    package_data_response = response.json()
+    required_json_data = get_required_json_data(package_data_response, package_url)
+    return required_json_data
+
+
+def scrape_conda(output_dir: str, letter_to_scrape: str) -> None:
+    """
+    Scrapes all the conda packages data for the given letter
+    and saves the data in a parquet file.
+
+    Args:
+        letter_to_scrape (str): The letter to scrape.
+
+    Returns:
+        None
+    """
+    output_dir = os.path.join("output", "conda")
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    package_data_list = list()
+    packages_name_list = get_all_package_names(letter_to_scrape)
+    for package_name in list(packages_name_list):
+        package_data = get_package_data(package_name)
+        if package_data:
+            package_data_list.append(package_data)
+    if package_data_list:
+        df = pd.DataFrame(package_data_list)
+        df.to_parquet(output_dir, partition_cols=["initial_letter"])
+        logger.info("Data saved for package : %s", len(package_data_list))
+    else:
+        logger.error("Failed to save data for package letter : %s", letter_to_scrape)
```

Comment:
this should no longer be required
```suggestion

    df = pd.DataFrame(package_data_list)
    df.to_parquet(output_dir, partition_cols=["initial_letter"])
    logger.info("Data saved for package : %s", len(package_data_list))
```

Reply:
This is updated now